File: backend/services/notification-service/src/main/java/com/lmspilot/notification/config/DevelopmentSeed.java
package com.lmspilot.notification.config;

import com.lmspilot.notification.domain.*;
import java.time.Instant;
import java.util.UUID;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.*;

@Configuration
public class DevelopmentSeed {
    @Bean
    CommandLineRunner seedNotification(NewsArticleRepository news, NotificationRepository notifications,
                                       @Value("${lmspilot.seed-demo:true}") boolean enabled) {
        return args -> {
            if (!enabled) return;

            UUID adminId = UUID.fromString("00000000-0000-0000-0000-000000000001");
            UUID studentId = UUID.fromString("00000000-0000-0000-0000-000000000003");

            if (news.count() == 0) {
                NewsArticleEntity n1 = new NewsArticleEntity();
                n1.title = "Chào mừng bạn đến với Hệ thống LMSPilot 0.21.0";
                n1.summary = "Hệ thống nâng cấp toàn diện với hiệu năng vượt trội";
                n1.content = "<p>Chào mừng bạn đến với phiên bản LMSPilot 0.21.0 chạy trên nền tảng Java 21 LTS và Spring Boot 3.5!</p>";
                n1.status = NewsStatus.PUBLISHED;
                n1.audienceType = NewsAudienceType.SYSTEM;
                n1.pinned = true;
                n1.priority = 10;
                n1.createdBy = adminId;
                news.save(n1);
            }

            if (notifications.count() == 0) {
                NotificationEntity notif = new NotificationEntity();
                notif.userId = studentId;
                notif.title = "Khóa học mới đã được giao";
                notif.body = "Bạn đã được phân công tham gia khóa học: Bắt đầu với LMSPilot 0.21.0";
                notif.type = "SYSTEM";
                notif.read = false;
                notif.createdAt = Instant.now();
                notifications.save(notif);
            }
        };
    }
}

File: backend/services/notification-service/src/main/kotlin/com/lmspilot/notification/config/DevelopmentSeed.kt
package com.lmspilot.notification.config

import com.lmspilot.notification.domain.NotificationChannel
import com.lmspilot.notification.domain.NotificationEntity
import com.lmspilot.notification.domain.NotificationRepository
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Component
class DevelopmentSeed(
    private val notifications: NotificationRepository,
    @Value("\${lmspilot.seed-demo:false}") private val enabled: Boolean,
) : ApplicationRunner {
    @Transactional
    override fun run(args: ApplicationArguments) {
        if (!enabled) return
        seed(
            eventId = UUID.fromString("00000000-0000-0000-0000-000000000401"),
            userId = UUID.fromString("00000000-0000-0000-0000-000000000001"),
            title = "Hệ thống demo đã sẵn sàng",
            body = "Bạn có thể kiểm tra người dùng, khóa học, lớp và báo cáo mẫu.",
        )
        seed(
            eventId = UUID.fromString("00000000-0000-0000-0000-000000000402"),
            userId = UUID.fromString("00000000-0000-0000-0000-000000000002"),
            title = "Bạn được phân công lớp mẫu",
            body = "Lớp làm quen LMSPilot đang mở trong phạm vi giảng dạy của bạn.",
        )
        seed(
            eventId = UUID.fromString("00000000-0000-0000-0000-000000000403"),
            userId = UUID.fromString("00000000-0000-0000-0000-000000000003"),
            title = "Bạn đã được ghi danh",
            body = "Khóa học làm quen LMSPilot đã xuất hiện trong trang học tập cá nhân.",
        )
    }

    private fun seed(eventId: UUID, userId: UUID, title: String, body: String) {
        if (!notifications.existsBySourceEventIdAndUserIdAndChannel(eventId, userId, NotificationChannel.IN_APP)) {
            notifications.save(
                NotificationEntity(
                    id = eventId,
                    sourceEventId = eventId,
                    userId = userId,
                    type = "DEMO",
                    title = title,
                    body = body,
                )
            )
        }
    }
}
